python/taichi/core/util.py

```diff
@@ -131,14 +157,24 @@ def format(all=False, diff=None):
                 Path(os.path.join(tc.get_repo_directory(), d)).rglob('*'))
     else:
         if diff is None:
-            files = repo.index.diff('HEAD')
+            # Finds all modified files from upstream/master to working tree
+            # 1. diffs between the index and upstream/master. Also inclulde
+            # origin/master for repo owners.
+            files = repo.index.diff('upstream/master')
```

Comment:
Should we ask the user to add `upstream` to their git remotes, if git finds nothing here?

Reply:
I named `taichi-dev` as `father` instead of `upsteam`, and `archibate` as `origin`, it's up to user's choice..

Reply:
i think `upstream` and `origin` are the common practices for how you name your remote repos. Maybe we can support things like `.taichi` to customize these bits. For now, I do hope people could follow common practices -- Just like why we would have code styling, it's the cheapest way for us all to collaborate effectively.

Reply:
added a TODO

Reply:
> i think `upstream` and `origin` are the common practices for how you name your remote repos. Maybe we can support things like `.taichi` to customize these bits. For now, I do hope people could follow common practices -- Just like why we would have code styling, it's the cheapest way for us all to collaborate effectively.

Yeah, I agree it's important to follow common practices. I think the updated implementation looks fine. As long as `ti format` does not crash when someone doesn't use `upstream`, we are good.